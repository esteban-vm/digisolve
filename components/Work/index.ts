export { default as Work, type WorkProps } from './Work'
