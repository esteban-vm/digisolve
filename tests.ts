import type { ReactElement } from 'react'
import { create, type ReactTestRendererJSON, type TestRendererOptions } from 'react-test-renderer'

const customCreate = (ui: ReactElement, options?: TestRendererOptions) => {
  return create(ui, options).toJSON() as ReactTestRendererJSON
}

export { customCreate as create }
export * from '@testing-library/react'
export { default as userEvent } from '@testing-library/user-event'
