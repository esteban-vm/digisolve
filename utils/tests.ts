import type { ReactElement } from 'react'
import { create, type TestRendererOptions } from 'react-test-renderer'

const customCreate = (ui: ReactElement, options?: TestRendererOptions) => {
  return create(ui, options).toJSON()
}

export { customCreate as create }
export * from '@testing-library/react'
export { default as userEvent } from '@testing-library/user-event'
