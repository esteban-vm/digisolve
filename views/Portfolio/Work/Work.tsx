import type { ImageComponent } from '@/types'
import Image from 'next/image'
import { styled } from '@/styles'

const WorkComponent: ImageComponent = ({ src, alt, ...rest }) => {
  return (
    <figure {...rest}>
      <Image src={src} alt={alt} className='work__image' />
      <p className='work__paragraph'>{alt}</p>
    </figure>
  )
}

const Work = styled(WorkComponent)`
  background-color: var(--color-black);
  overflow: hidden;

  .work {
    &__image {
      width: 100%;
      height: auto;
      opacity: 0.7;
      transform: scale(1.15);
      transition: transform 500ms, opacity 500ms;

      :hover {
        transform: scale(1);
        opacity: 1;
      }
    }

    &__paragraph {
      color: var(--color-white);
      text-transform: capitalize;
    }
  }
`

export default Work
