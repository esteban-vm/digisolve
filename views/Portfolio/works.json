[
  {
    "title": "Online advertising",
    "image": {
      "src": "/img/online-advertising.png",
      "height": 470,
      "width": 394
    }
  },
  {
    "title": "Social media campaigns",
    "image": {
      "src": "/img/social-media-campaigns.png",
      "height": 470,
      "width": 394
    }
  },
  {
    "title": "UX design",
    "image": {
      "src": "/img/ux-design.png",
      "height": 470,
      "width": 394
    }
  },
  {
    "title": "Branding & digital marketing",
    "image": {
      "src": "/img/branding-digital-marketing.png",
      "height": 470,
      "width": 394
    }
  }
]
