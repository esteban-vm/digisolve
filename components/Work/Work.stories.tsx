import type { Meta, StoryObj } from '@storybook/react'
import { screen, userEvent } from '@storybook/testing-library'
import { advertising, campaigns } from '@/utils/images'
import { styled } from '@/utils/styles'
import Work from './Work'

const Wrapper = styled.div`
  text-align: center;
  height: auto;
  width: 100%;
`

export default {
  argTypes: {
    image: {
      control: false,
    },
    onClick: {
      action: true,
      control: false,
    },
    onMouseEnter: {
      action: true,
      control: false,
    },
  },
  component: Work,
  title: 'Components/Work',
  decorators: [
    (Story, context) => {
      return (
        <Wrapper>
          <Story {...context} />
        </Wrapper>
      )
    },
  ],
  parameters: {
    layout: 'centered',
  },
  play() {
    const img = screen.getByRole('img')
    userEvent.click(img)
    userEvent.hover(img)
  },
} as Meta<typeof Work>

export const WorkExample1: StoryObj<typeof Work> = {
  args: {
    image: advertising,
    text: 'Online advertising',
  },
}

export const WorkExample2: StoryObj<typeof Work> = {
  args: {
    image: campaigns,
    text: 'Social media campaigns',
  },
}
