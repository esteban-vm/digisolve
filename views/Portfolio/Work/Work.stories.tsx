import { type Meta, type StoryObj, within, userEvent } from '@/stories'
import { styled } from '@/styles'
import { works } from '../Portfolio'
import Work from './Work'

const Wrapper = styled.div`
  text-align: center;
`

export default {
  argTypes: {
    src: {
      control: false,
    },
  },
  component: Work,
  title: 'Views/Portfolio/Works',
  decorators: [
    (Story, context) => {
      return (
        <Wrapper>
          <Story {...context} />
        </Wrapper>
      )
    },
  ],
  parameters: {
    layout: 'centered',
  },
  play({ canvasElement }) {
    const canvas = within(canvasElement)
    const image = canvas.getByRole('img')
    userEvent.click(image)
    userEvent.hover(image)
  },
} as Meta<typeof Work>

const Examples: StoryObj<typeof Work>[] = []
for (const work of works) Examples.push({ args: work })

export const Example1 = Examples[0]
export const Example2 = Examples[1]
export const Example3 = Examples[2]
