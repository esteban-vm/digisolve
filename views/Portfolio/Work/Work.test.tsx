import { render, cleanup, screen, create } from '@/tests'
import works from '../Portfolio.data'
import Work from './Work'

describe('ðŸ§ª WORK:', () => {
  const [testWork] = works

  describe('display tests:', () => {
    beforeEach(() => {
      render(<Work {...testWork} />)
    })

    afterEach(cleanup)

    it('should be accessible', () => {
      const figure = screen.getByRole('figure', { name: testWork.text })
      expect(figure).toBeInTheDocument()
    })

    describe('should be displayed:', () => {
      it('the image', () => {
        const image = screen.getByRole('img', { name: testWork.text })
        expect(image).toBeInTheDocument()
        expect(image).toBeVisible()
      })

      it('the title', () => {
        const title = screen.getByText(testWork.text)
        expect(title).toBeInTheDocument()
        expect(title).toBeVisible()
      })
    })
  })

  describe('style tests:', () => {
    let tree: ReturnType<typeof create>

    beforeEach(() => {
      tree = create(<Work {...testWork} />)
    })

    it('should render with correct styles', () => {
      expect(tree).toMatchSnapshot()
    })

    it('should have correct styles when hovered', () => {
      const target = 'img:hover'
      expect(tree).toHaveStyleRule('transform', 'scale(1)', { target })
      expect(tree).toHaveStyleRule('opacity', '1', { target })
    })
  })
})
