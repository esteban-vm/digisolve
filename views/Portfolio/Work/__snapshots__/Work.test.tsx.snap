// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ðŸ§ª WORK: style tests: should render with correct styles 1`] = `
.emotion-0 {
  background-color: var(--color-black);
  overflow: hidden;
}

.emotion-0 img {
  width: 100%;
  height: auto;
  opacity: 0.7;
  -webkit-transform: scale(1.15);
  -moz-transform: scale(1.15);
  -ms-transform: scale(1.15);
  transform: scale(1.15);
  -webkit-transition: -webkit-transform 500ms,opacity 500ms;
  transition: transform 500ms,opacity 500ms;
}

.emotion-0 img:hover {
  -webkit-transform: scale(1);
  -moz-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  opacity: 1;
}

.emotion-0 p {
  color: var(--color-white);
  text-transform: capitalize;
}

<figure
  aria-labelledby="0a6dd16e-0229-4898-a3b4-b6b8a2dd5e5a"
  className="emotion-0"
>
  <img
    alt="Online advertising"
    data-nimg="1"
    decoding="async"
    height={470}
    id="0a6dd16e-0229-4898-a3b4-b6b8a2dd5e5a"
    loading="lazy"
    onError={[Function]}
    onLoad={[Function]}
    src="/_next/image?url=%2Fimg%2Fonline-advertising.png&w=828&q=75"
    srcSet="/_next/image?url=%2Fimg%2Fonline-advertising.png&w=640&q=75 1x, /_next/image?url=%2Fimg%2Fonline-advertising.png&w=828&q=75 2x"
    style={
      {
        "color": "transparent",
      }
    }
    width={394}
  />
  <p>
    Online advertising
  </p>
</figure>
`;
